### PR TITLE
DDPB-2512 - NDR report incorrectly displayed

### PR DESCRIPTION
**Pull request by acsauk-mojd**
I've moved the NDR creation step to the Lay Deputy specific steps during the upsert action to ensure they are only created for Lay Deputies.

As I was running the unit tests locally I encountered an error while loading the fixtures class so I've namespaced this and updated references as part of this PR.

### CODE REVIEW

File: src/AppBundle/Controller/ClientController.php [hunk 1]
```diff
@@ -49,6 +49,13 @@ public function upsertAction(Request $request)
         ]);
 
         if ($this->getUser()->isLayDeputy()) {
+            //add NDR if not added yet
+            // TODO move to listener or service
```
Comment (aiesoftware) [inline src/AppBundle/Controller/ClientController.php]: I think we should remove these comments..

**Comment by acsauk-mojd** [comment]
The requirements of this ticket evolved slightly - NDRs should only be created for Lays if the NDR Enabled checkbox has been ticked from the admin side. We were previously just checking if an NDR existed for the client rather than checking for the box being checked. I've updated the logic to reflect this and added a couple fo extra assertions for this in the unit tests.

File: src/AppBundle/Controller/ClientController.php [hunk 2]
```diff
@@ -60,14 +71,8 @@ public function upsertAction(Request $request)
...
 
-        $this->persistAndFlush($client);
-
-        //add NDR if not added yet
-        // TODO move to listener or service
-        if (!$client->getNdr()) {
-            $ndr = new EntityDir\Ndr\Ndr($client);
-            $this->persistAndFlush($ndr);
-        }
+        $em->persist($client);
```
Comment (slizzio) [inline src/AppBundle/Controller/ClientController.php]: check client is set before persisting

File: tests/AppBundle/Controller/ClientControllerTest.php [hunk 3]
```diff
@@ -121,10 +121,9 @@ public function testupsertPostLayDeputy()
         $this->assertEquals('Firstname', $client->getFirstname());
         $this->assertCount(1, $client->getUsers());
         $this->assertEquals(self::$deputy1->getId(), $client->getUsers()->first()->getId());
-        $this->assertInstanceOf('AppBundle\Entity\Ndr\Ndr', $client->getNdr());
     }
 
-    public function testupsertPut()
+    public function testupsertPut_lay_deputy()
```
Comment (slizzio) [inline tests/AppBundle/Controller/ClientControllerTest.php]: camelCase?

File: tests/AppBundle/Controller/ClientControllerTest.php [hunk 4]
```diff
@@ -148,6 +147,30 @@ public function testupsertPut()
         $this->assertEquals(null, $client->getDateOfBirth());
         $this->assertEquals('2015-12-31', $client->getCourtDate()->format('Y-m-d'));
         $this->assertEquals(self::$deputy1->getId(), $client->getUsers()->first()->getId());
+    }
+
+    public function testupsertPut_lay_deputy_ndr_enabled()
```
Comment (slizzio) [inline tests/AppBundle/Controller/ClientControllerTest.php]: camelCase

File: tests/AppBundle/Controller/ClientControllerTest.php [hunk 5]
```diff
@@ -148,6 +147,30 @@ public function testupsertPut()
...
+            'mustSucceed' => true,
+            'AuthToken' => self::$tokenDeputy,
+            'data' => ['id' => self::$client1->getId(), 'ndr_enabled' => true] + $this->updateDataLay,
+        ]);
+        self::fixtures()->clear();
+        $client = self::fixtures()->getRepo('Client')->find($return['data']['id']); /* @var $client \AppBundle\Entity\Client */
+        $this->assertInstanceOf(Ndr::class, $client->getNdr());
+    }
+
+    public function testupsertPut_PA()
```
Comment (slizzio) [inline tests/AppBundle/Controller/ClientControllerTest.php]: camelcase

File: tests/AppBundle/Controller/ClientControllerTest.php (hunk 3 above)
Comment (acsauk-mojd) [reply tests/AppBundle/Controller/ClientControllerTest.php]: I used to be exclusively camel case for test function names until my last role where we kept the actual function it was testing in camel case but gave the scenario/detail using snake case. Happy to change if thats the preferred style here though!

**Pull request merged**